src/roswire/ros2/node_manager.py

```diff
@@ -1,30 +1,42 @@
 # -*- coding: utf-8 -*-
 __all__ = ('ROS2NodeManager',)
 
-from typing import Iterator, Mapping
+from typing import Iterator
 import typing
+import attr
 
 from loguru import logger
-import attr
 
 from .node import ROS2Node
+from .state import ROS2StateProbe
+from ..proxy import SystemState
 from .. import exceptions as exc
+from ..interface import NodeManager
 
 if typing.TYPE_CHECKING:
     from ..app import AppInstance
 
 
 @attr.s(frozen=True, auto_attribs=True, slots=True)
-class ROS2NodeManager(Mapping[str, ROS2Node]):
+class ROS2NodeManager(NodeManager):
     """Provides an interface for interacting with ROS2 nodes."""
-    app_instance: 'AppInstance'
+    app_instance: 'AppInstance' = attr.ib()
+    _state_probe: 'ROS2StateProbe' = attr.ib(init=False)
+
+    def __attrs_post_init__(self) -> None:
+        _state_probe = ROS2StateProbe.for_app_instance(self.app_instance)
+        object.__setattr__(self, 'state_probe', _state_probe)
```

Comment:
The second argument, `'state_probe'`, gives the name of the attribute that should be set as a string. In this case, we want to set the `_state_probe` attribute, so we should add the `_` prefix to the string. Note that the local variable `_state_probe` does not need a leading underscore (it can't be accessed outside of the scope of this method).